lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2746,6 +2750,16 @@ ResolveWitnessResult ConformanceChecker::resolveTypeWitnessViaLookup(
     }
   }
 
+  // If there are no viable witnesses, and all nonviable candidates came from
+  // protocol extensions, treat this as "missing".
+  if (viable.empty() &&
+      std::find_if(nonViable.begin(), nonViable.end(),
+                   [](const std::pair<TypeDecl *, CheckTypeWitnessResult> &x) {
+                     return x.first->getDeclContext()
+                        ->getAsProtocolOrProtocolExtensionContext() == nullptr;
```

Comment:
The comment at the top mentioned "protocol extensions", so I'm wondering is it _only_ protocol extensions? If so I think we should be using getAsProtocolExtensionContext() ?

Reply:
Typealiases in protocols would be okay here, too; if we see them along this path, it's fine to treat them the same way as typealiases in protocol extensions.

Reply:
Gotcha, thanks!